enterprise-4.13 OADP-4657: Release Notes for 1.4.0 for OCP 4.13 (#80585)
openshift/openshift-docs: modules/oadp-1-4-0-release-notes.adoc

:_newdoc-version: 2.18.3
:_template-generated: 2024-06-28
// Module included in the following assemblies:
//

> // * backup_and_restore/oadp-1-4-release-notes.adoc

Reviewer: This seems like a required change to me

:_mod-docs-content-type: REFERENCE

[id="oadp-1-4-0-release-notes_{context}"]
= OADP 1.4.0 release notes

> The {oadp-first} 1.4.0 release notes lists resolved issues and known issues.

Reviewer: Subject/ Verb agreement required change. See https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#general-guidelines

[id="resolved-issues-1-4-0_{context}"]
== Resolved issues

.Restore works correctly in {product-title} 4.16

> Previously, while restoring the deleted application namespace, the restore operation partially failed with the `resource name may not be empty` error in {product-title} 4.16.

Reviewer: This seems like more of a suggestion. I think the way Andy has the error message inline is OK too. I can't find guidance explicitly saying not to do it like this.

> With this update, restore works as expected in {product-title} 4.16.

Reviewer: Adding an article: https://www.ibm.com/docs/en/ibm-style?topic=grammar-articles

link:https://issues.redhat.com/browse/OADP-4075[OADP-4075]

.Data Mover backups work properly in the {product-title} 4.16 cluster

> Previously, Velero was using the earlier version of SDK where the `Spec.SourceVolumeMode` field did not exist. As a consequence, Data Mover backups failed in the {product-title} 4.16 cluster on the external snapshotter with version 4.2.

Reviewer: - used is a valid minimalism alternative to "was using"
- I think maybe this should be the Operator SDK?

link:https://issues.redhat.com/browse/OADP-3922[OADP-3922]

For a complete list of all issues resolved in this release, see the list of link:https://issues.redhat.com/issues/?filter=12438505[OADP 1.4.0 resolved issues] in Jira.

[id="known-issues-1-4-0_{context}"]
== Known issues

.Backup fails when checksumAlgorithm is not set for MCG

While performing a backup of any application with Noobaa as the backup location, if the `checksumAlgorithm` configuration parameter is not set, backup fails. To fix this problem, if you do not provide a value for `checksumAlgorithm` in the Backup Storage Location (BSL) configuration, an empty value is added.
The empty value is only added for BSLs that are created using Data Protection Application (DPA) custom resource (CR), and this value is not added if BSLs are created using any other method.
link:https://issues.redhat.com/browse/OADP-4274[OADP-4274]

For a complete list of all known issues in this release, see the list of link:https://issues.redhat.com/issues/?filter=12438506[OADP 1.4.0 known issues] in Jira.

[id="upgrade-notes-1-4-0_{context}"]
== Upgrade notes

> [NOTE]

Reviewer: Since this seems like a required step, changing from a NOTE to IMPORTANT seems appropriate here: https://redhat-documentation.github.io/supplementary-style-guide/#admonitions

====
Always upgrade to the next minor version. *Do not* skip versions. To update to a later version, upgrade only one channel at a time. For example, to upgrade from {oadp-first} 1.1 to 1.3, upgrade first to 1.2, and then to 1.3.
====

[id="changes-oadp-1-3-to-1-4_{context}"]

> === Changes from OADP 1.3 to 1.4

Reviewer 2: Can we make this a concept module?

> The Velero server has been updated from version 1.12 to 1.14. Note that there are no changes in the Data Protection Application (DPA).

Reviewer: ![image](https://github.com/user-attachments/assets/03d341f3-d57c-4273-87be-b389a61f7626)

IBMSG: To upgrade is to replace software with a new release or fix level of the same program or to replace hardware with a newer model or more powerful technology. An upgrade involves little work on the user’s part.

> This changes the following:

Reviewer: I agree, being more specific that the update changes the following items is clearer.

> * The `velero-plugin-for-csi` code is now available in the Velero code, which means an `init` container is no longer required for the plugin.

Reviewer 2: ```suggestion
* The `velero-plugin-for-csi` code is now available in the Velero code, which means that an `init` container is no longer needed for the plugin.
```

> * Velero changed client Burst and QPS defaults from 30 and 20 to 100 and 100, respectively.

Reviewer 2: ```suggestion
* Velero has changed the default values for client burst and QPS from 30 and 20 to 100 and 100 respectively.
```

> * The `velero-plugin-for-aws` plugin updated default value of the `spec.config.checksumAlgorithm` field in `BackupStorageLocation` objects (BSLs) from `""` (no checksum calculation) to the `CRC32` algorithm. For more information, see link:https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/release-1.10/backupstoragelocation.md[Velero plugins for AWS Backup Storage Location]. The checksum algorithm types are known to work only with AWS.

Reviewer 2: ```suggestion
* The `velero-plugin-for-aws` plugin has updated the default value of the `spec.config.checksumAlgorithm` field in `BackupStorageLocation` objects (BSLs) from `""` (no checksum calculation) to the `CRC32` algorithm. 
+
For more information, see link:https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/release-1.10/backupstoragelocation.md[Velero plugins for AWS Backup Storage Location]. 
+
[NOTE]
====
The checksum algorithms only work with AWS.
====
```

Reviewer 2: Expand AWS??

Reviewer: It looks like this is the first mention of AWS on the page, just by a cursory glance. If that's the case then IBMSG recommends spelling out first mention: https://www.ibm.com/docs/en/ibm-style?topic=references-product-service-names#service-names

> * The `velero-plugin-for-aws` plugin updated default value of the `spec.config.checksumAlgorithm` field in `BackupStorageLocation` objects (BSLs) from `""` (no checksum calculation) to the `CRC32` algorithm. For more information, see link:https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/release-1.10/backupstoragelocation.md[Velero plugins for AWS Backup Storage Location]. The checksum algorithm types are known to work only with AWS.

Reviewer: We cannot link to GitHub, per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-to-external-websites.

> Several S3 providers require the `md5sum` to be disabled by setting the checksum algorithm to `""`. Confirm `md5sum` algorithm support and configuration with your storage provider.
> +
> In OADP 1.4, the default value for BSLs created within DPA for this configuration is `""`. This default value means that the `md5sum` is not checked, which is consistent with OADP 1.3. For BSLs created within DPA, update it by using the `spec.backupLocations[].velero.config.checksumAlgorithm` field in the DPA. If your BSLs are created outside DPA, you can update this configuration by using `spec.config.checksumAlgorithm` in the BSLs.

Reviewer 2: ```suggestion
For some S3 providers, you must disable the `md5sum`. To disable the `md5sum`, you must set the checksum algorithm to `""`. Confirm the `md5sum` support and configuration of the `md5sum` algorithm with your storage provider.
+
In OADP 1.4, the default value for BSLs created within DPA for this configuration is `""`. This default value means that the `md5sum` is not checked, which is consistent with OADP 1.3. For BSLs created within DPA, update it by using the `spec.backupLocations[].velero.config.checksumAlgorithm` field in DPA. If your BSLs are created outside of DPA, you can upgrade this configuration by using `spec.config.checksumAlgorithm` in the BSLs.
```
